pkg/controller/build/build_controller.go

```diff
@@ -123,17 +123,24 @@ func (r *ReconcileBuild) Reconcile(request reconcile.Request) (reconcile.Result,
 }
 
 func (r *ReconcileBuild) validateStrategyRef(s *build.StrategyRef, ns string) error {
-	switch *s.Kind {
-	case build.NamespacedBuildStrategyKind:
-		if err := r.validateBuildStrategy(s.Name, ns); err != nil {
-			return err
+	if s.Kind != nil {
+		switch *s.Kind {
+		case build.NamespacedBuildStrategyKind:
+			if err := r.validateBuildStrategy(s.Name, ns); err != nil {
+				return err
+			}
+		case build.ClusterBuildStrategyKind:
+			if err := r.validateClusterBuildStrategy(s.Name); err != nil {
+				return err
+			}
+		default:
+			return fmt.Errorf("unknown strategy %v", *s.Kind)
 		}
-	case build.ClusterBuildStrategyKind:
-		if err := r.validateClusterBuildStrategy(s.Name); err != nil {
+	} else {
+		log.Info("BuildStrategy kind is nil, use default NamespacedBuildStrategyKind")
+		if err := r.validateBuildStrategy(s.Name, ns); err != nil {
```

Comment:
I´m not sure we should set default to buildstrategy when the Kind is not defined. I would expect to return a proper error if the `BuildStrategyKind` is nil(not defined). Whats the reasoning behind setting a default? this is afaik an expected field the user needs to set, and if not set the user will get the error under the Build `REASON` column, see https://github.com/redhat-developer/build/blob/master/pkg/controller/build/build_controller.go#L112

Reply:
This is the disucssion before:
https://github.com/redhat-developer/build/pull/62#discussion_r391671438

Reply:
I read https://github.com/redhat-developer/build/pull/62#discussion_r391671438 , seems this behaviour when defaulting to a BuildStrategy is desired.